server/src/modules/post/post.controller.ts

```diff
@@ -159,17 +163,23 @@ export class PostController {
   /**
    * Get a single post and all the tags and users associated with it
    * @param postId Id of the post
+   * @query relatedPosts if true, return related posts
    * @return 200 with post
    * @return 403 if user does not have permission to access post
    * @return 500 for database error
    */
-  getSinglePost: ControllerHandler<{ id: string }, Post | Message> = async (
-    req,
-    res,
-  ) => {
+  getSinglePost: ControllerHandler<
+    { id: string },
+    PostWithUserTagRelations | PostWithUserTagRelatedPostRelations | Message,
+    undefined,
+    { noOfRelatedPosts?: string }
```

Comment:
I think we could get away with `relatedPosts` and make it a number, and it will be sensible enough. Do also consider adding the necessary validator to `post.routes`

Reply:
Added